main.ts

```diff
@@ -1,23 +1,37 @@
 import { App, ButtonComponent, Modal, Notice, Plugin, PluginSettingTab, Setting, TextComponent } from 'obsidian';
 import { createDailyNote } from 'obsidian-daily-notes-interface';
 
+interface IReviewSettings {
+	dailyNotesFolder: string;
+	reviewSectionHeading: string;
+	linePrefix: string;
+	defaultReviewDate: string;
+	blockLinePrefix: string;
+}
+
+const DEFAULT_SETTINGS: IReviewSettings = {
+	dailyNotesFolder: "",
+	reviewSectionHeading: "## Review",
+	linePrefix: "- ",
+	defaultReviewDate: "tomorrow",
+	blockLinePrefix: "!",
+}
+
+
 export default class Review extends Plugin {
-	settings: ReviewSettings;
+	settings: IReviewSettings;
 
 	async onload() {
 		console.log('Loading the Review plugin.');
 
-		// Check that plugins can be accessed.
-		console.log(app.plugins.plugins);
+		this.settings = Object.assign({}, DEFAULT_SETTINGS, (await this.loadData()))
 
-		// Check for the Natural Language Dates plugin. If not found, tell the user to install it/initialize it.
-		let naturalLanguageDates = app.plugins.getPlugin('nldates-obsidian');
-		if (!naturalLanguageDates) {
-			new Notice("The Natural Language Dates plugin was not found. The Review plugin requires the Natural Language Dates plugin. Please install it first and make sure it is enabled before using Review.");
+		if (this.app.workspace.layoutReady) {

```

Comment:
Can't check for nl-dates in onLoad since the load order is nondeterministic. Instead, wait until layout-ready event is fired to check for nl-dates.